src/jit/codegenxarch.cpp

```diff
@@ -1329,9 +1345,23 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
     }
     else
     {
-        // dividend must be in RAX
-        if (dividend->gtRegNum != REG_RAX)
+        if (dividendIsLong)
+        {
+            assert(dividendLo != nullptr && dividendHi != nullptr);
+
+            // dividendLo must be in RAX; dividendHi must be in RDX
+            if (dividendLo->gtRegNum != REG_RAX)
```

Comment:
LSRA tries as much as possible to allocate low and high parts to RAX and RDX.   But still it is not always a guarantee that the dividend (low and high parts) may not be in RAX/RDX.  What LSRA guarantees is that RAX and RDX are available for div/mod operation.  It is still codegen responsibility to make sure that dividend gets moved to right register if it is not already present in the right register.